src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,571 @@
+        be withdrawn from the pool.
+        @param ilk bytes32 of the D3M ilk name
+    */
+    function reap(bytes32 ilk) external {
+        D3MPoolLike pool = ilks[ilk].pool;
+
+        require(vat.live() == 1, "DssDirectDepositHub/no-reap-during-shutdown");
+        require(ilks[ilk].tic == 0, "DssDirectDepositHub/pool-not-live");
+
+        pool.accrueIfNeeded();
+        uint256 assetBalance = pool.assetBalance();
+        (, uint256 daiDebt) = vat.urns(ilk, address(pool));
+        if (assetBalance > daiDebt) {
+            uint256 fees = assetBalance - daiDebt;
+            fees = _min(fees, pool.maxWithdraw());
+            pool.withdraw(fees);
+            daiJoin.join(vow, fees);
+            emit Reap(ilk, fees);
+        }
+    }
+
+    /**
+        @notice Allow Users to return vat gem for Pool Shares.
+        This will only occur during Global Settlement when users receive
+        collateral for their Dai.
+        @param ilk bytes32 of the D3M ilk name
+        @param usr address that should receive the shares from the pool
+        @param wad amount of gems that the msg.sender is returning
+    */
+    function exit(bytes32 ilk, address usr, uint256 wad) external {
+        require(wad <= 2 ** 255, "DssDirectDepositHub/overflow");
+        vat.slip(ilk, msg.sender, -int256(wad));
+        D3MPoolLike pool = ilks[ilk].pool;
+        require(pool.transfer(usr, wad), "DssDirectDepositHub/failed-transfer");
+        emit Exit(ilk, usr, wad);
+    }
+
+    /**
+        @notice Shutdown a pool.
+        This starts the countdown to when the debt can be written off (cull).
+        Once called, subsequent calls to `exec` will unwind as much of the
+        position as possible.
+        @dev msg.sender must be authorized.
+        @param ilk bytes32 of the D3M ilk name
+    */
+    function cage(bytes32 ilk) external auth {
+        require(vat.live() == 1, "DssDirectDepositHub/no-cage-during-shutdown");
+
+        ilks[ilk].tic = block.timestamp;
+        emit Cage(ilk);
+    }
+
+    /**
+        @notice Write off the debt for a caged pool.
+        This must occur while vat is live and after tau number of seconds has
+        passed since the pool was caged.
+        @dev This will send the pool's debt to the vow as sin and convert its
+        collateral to gems.  There is a situation where another user has paid
+        back some of the Pool's debt where ink != art in this case we rebalance
+        so that vat.gems(pool) will represent the amount of debt sent to the vow.
+        @param ilk bytes32 of the D3M ilk name
+    */
+    function cull(bytes32 ilk) external {
+        require(vat.live() == 1, "DssDirectDepositHub/no-cull-during-shutdown");
+
+        uint256 tic = ilks[ilk].tic;
+        require(tic > 0, "DssDirectDepositHub/pool-live");
+
+        uint256 tau = ilks[ilk].tau;
+        require(_add(tic, tau) <= block.timestamp || wards[msg.sender] == 1, "DssDirectDepositHub/unauthorized-cull");
```

Comment:
Doing things this ways allows modifications of `tau` after an ilk has been caged to affect the time-to-`cull`. Is that an intentional design choice? If not, would we prefer to capture the original `tau` value in `tic` so that this check would be just `tic <= block.timstamp`?